lib/components/admin/print-field-trip-layout.js

```diff
@@ -0,0 +1,184 @@
+      // for fetching/receiving the details of the field trip per request id.
+      receivedFieldTrips({
+        fieldTrips: [{
+          endTime: 0,
+          id: requestId
+        }]
+      })
+      fetchFieldTripDetails(requestId)
+    }
+
+    if (request && request !== prevProps.request) {
+      // Set window title when request has fully loaded
+      // (appears in print headings)
+      const { endLocation, schoolName } = request
+      document.title = `Field Trip: ${schoolName} to ${endLocation}`
+    }
+  }
+
+  componentWillUnmount () {
+    clearClassFromRootHtml()
+  }
+
+  render () {
+    const { config, request } = this.props
+    const { LegIcon } = this.context
+    if (!request) return null
+
+    const {
+      address,
+      classpassId,
+      emailAddress,
+      endLocation,
+      faxNumber,
+      grade,
+      numChaperones,
+      numFreeStudents,
+      numStudents,
+      phoneNumber,
+      schoolName,
+      teacherName,
+      timeStamp
+    } = request
+
+    // Outbound/inbound template
+    const tripStructure = [
+      {
+        title: 'Outbound Trip (to Destination)',
+        trip: getTripFromRequest(request, true),
+        tripAbsentMessage: 'No Outbound Trip Planned'
+      },
+      {
+        title: 'Inbound Trip (from Destination)',
+        trip: getTripFromRequest(request, false),
+        tripAbsentMessage: 'No Inbound Trip Planned'
+      }
+    ]
+
+    return (
+      <PrintLayout>
+        <Header>
+          <Button bsSize='small' onClick={this._print} style={{ float: 'right' }}>
+            <i className='fa fa-print' /> Print
+          </Button>
+          <TripTitle>Field Trip Plan: {schoolName} to {endLocation}</TripTitle>
+        </Header>
+        <TripInfoList>
+          <li><b>Teacher</b>: <Val>{teacherName}</Val> ({schoolName}, Grade: <Val>{grade}</Val>)</li>
+          <li><b>Teacher Address</b>: <Val>{address}</Val></li>
+          <li><b>Phone</b>: <Val>{phoneNumber}</Val> / <b>Fax</b>: <Val>{faxNumber}</Val></li>
+          <li><b>Email</b>: <Val>{emailAddress}</Val></li>
+          <li><b>Students Age 7 and Over</b>: {numStudents || 0}</li>
+          <li><b>Students Age 6 and Under</b>: {numFreeStudents || 0}</li>
+          <li><b>Chaperones</b>: {numChaperones || 0}</li>
+          {classpassId && <li><b>Class Pass Hop Card #</b>: {classpassId}</li>}
+          <li><i>Request submitted: {timeStamp}</i></li>
+        </TripInfoList>
+
+        {tripStructure.map(({ title, trip, tripAbsentMessage }, i) => (
+          <TripContainer key={i}>
+            <h2>{title}</h2>
+            {trip
+              ? trip.groupItineraries?.map((groupItin, i) => {
+                const itinerary = JSON.parse(lzwDecode(groupItin.itinData))
+                return (
+                  <TripBody key={i}>
+                    <ItineraryContainer>
+                      <h3>{groupItin.passengers} passengers on following itinerary:</h3>
+                      <ItineraryBody>
```

Comment:
I got a little confused by this since we use `itinerary-body` to render an itinerary. Maybe rename it `ItineraryBodyContainer`?

Reply:
I used the class name in the old client that's why. Renamed in https://github.com/opentripplanner/otp-react-redux/pull/398/commits/6ef72b9b8d00366f5c62717405be677c0bb34233.